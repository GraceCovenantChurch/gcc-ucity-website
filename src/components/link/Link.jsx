import React, { forwardRef } from "react";
import PropTypes from "prop-types";
import { Link as RouterLink } from "react-router-dom";

import isExternal from "is-url-external";

const Link = forwardRef((props, ref) => {
  const to = props.to;

  return isExternal(to) ? (
    <a ref={ref} href={to} {...props}>
      {props.children}
    </a>
  ) : (
    <RouterLink {...props} />
  );
});

Link.propTypes = {
  to: PropTypes.string.isRequired,
};

export default Link;
