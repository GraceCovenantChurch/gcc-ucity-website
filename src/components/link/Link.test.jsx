import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

import Link from "./Link";

describe("Link", () => {
  it("renders correctly with internal", () => {
    const tree = renderer
      .create(
        <Router>
          <Link to="/home">
            <div />
          </Link>
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with external", () => {
    const tree = renderer
      .create(
        <Router>
          <Link to="https://www.google.com">
            <div />
          </Link>
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
