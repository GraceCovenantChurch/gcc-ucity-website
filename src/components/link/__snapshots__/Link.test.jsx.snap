// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly with external 1`] = `
<a
  className="makeStyles-link-1"
  href="https://www.google.com"
  to="https://www.google.com"
>
  <div />
</a>
`;

exports[`Link renders correctly with internal 1`] = `
<a
  className="makeStyles-link-1"
  href="/home"
  onClick={[Function]}
>
  <div />
</a>
`;
